### PR TITLE
Add gzip (compression) settings to lucky apps

### PR DESCRIPTION
**Pull request by bdtomlin**
Adding pre-compression of static assets to Lucky projects as referenced in https://github.com/luckyframework/lucky/issues/154

Notes:

* ~~There are failing tests in the master branch that continue to fail in this branch.~~ apparently this is just local for me.
* This currently uses Lucky from the branch in my repo that is referenced in this pull request: https://github.com/luckyframework/lucky/pull/982 since a new constant must be referenced by the app in the tests.

### CODE REVIEW

**Comment by bdtomlin** [comment]
Relate to https://github.com/luckyframework/lucky/pull/983 as https://github.com/luckyframework/lucky/pull/982 has now been merged into 983.

File: src/browser_app_skeleton/package.json [hunk 1]
```diff
@@ -14,10 +14,11 @@
...
     "browser-sync": "^2.18.13",
+    "compression-webpack-plugin": "^3.0.0",
     "laravel-mix": "^4.0.0",
     "resolve-url-loader": "2.3.1",
     "sass": "1.17.1",
     "sass-loader": "7.*",
     "vue-template-compiler": "^2.5.22"
   }
-}
+}
```
Comment (paulcsmith) [inline src/browser_app_skeleton/package.json]: Could you add a newline here?

File: src/browser_app_skeleton/webpack.mix.js [hunk 2]
```diff
@@ -6,15 +6,35 @@
...
+if (mix.inProduction()) {
+  let CompressionWepackPlugin = require('compression-webpack-plugin');
+  let gzipCompression = new CompressionWepackPlugin({
+    compressionOptions: { level: 9 },
+    test: /\.js$|\.css$|\.html$|\.svg$/
+  })
+  plugins.push(gzipCompression)
+  // add additional compression plugins here. For example
+  // if you want to add brotli compression...
+  // let brotliCompression = new CompressionWepackPlugin({
```
Comment (paulcsmith) [inline src/browser_app_skeleton/webpack.mix.js]: Could you add an empty line above this so it is clear where the code starts?

File: src/browser_app_skeleton/webpack.mix.js [hunk 3]
```diff
@@ -6,15 +6,35 @@
...
+
+// Compress static assets in production
+if (mix.inProduction()) {
+  let CompressionWepackPlugin = require('compression-webpack-plugin');
+  let gzipCompression = new CompressionWepackPlugin({
+    compressionOptions: { level: 9 },
+    test: /\.js$|\.css$|\.html$|\.svg$/
+  })
+  plugins.push(gzipCompression)
+  // add additional compression plugins here. For example
```
Comment (paulcsmith) [inline src/browser_app_skeleton/webpack.mix.js]:
```suggestion
  // Add additional compression plugins here. For example
```

File: src/browser_app_skeleton/webpack.mix.js [hunk 4]
```diff
@@ -6,15 +6,35 @@
...
+  // if you want to add brotli compression...
```
Comment (paulcsmith) [inline src/browser_app_skeleton/webpack.mix.js]:
```suggestion
  // if you want to add Brotli compression...
```

File: src/web_app_skeleton/config/server.cr.ecr [hunk 5]
```diff
@@ -7,6 +7,12 @@ Lucky::Server.configure do |settings|
     settings.secret_key_base = secret_key_from_env
     settings.host = "0.0.0.0"
     settings.port = ENV["PORT"].to_i
+    settings.gzip_enabled = true
+    # by default certain content types will be gzipped.
```
Comment (paulcsmith) [inline src/web_app_skeleton/config/server.cr.ecr]:
```suggestion
    # By default certain content types will be gzipped.
```

File: src/web_app_skeleton/config/server.cr.ecr [hunk 6]
```diff
@@ -7,6 +7,12 @@ Lucky::Server.configure do |settings|
...
+    # For a full list look in
+    # https://github.com/luckyframework/lucky/blob/master/src/lucky/server.cr
+    # to add additional extensions do something like this:
```
Comment (paulcsmith) [inline src/web_app_skeleton/config/server.cr.ecr]:
```suggestion
    # To add additional extensions do something like this:
```

And maybe a newline above this to separate it from the link a bit more

File: src/browser_app_skeleton/webpack.mix.js (hunk 2 above)
Comment (bdtomlin) [reply src/browser_app_skeleton/webpack.mix.js]: I think this is what you mean...

Comment (paulcsmith) [reply src/browser_app_skeleton/webpack.mix.js]: Yep looks good!

**Pull request merged**